scripts/xtensa-build-zephyr.py

```diff
@@ -899,6 +901,60 @@ def build_platforms():
 			symlinks=True, ignore_dangling_symlinks=True, dirs_exist_ok=True)
 
 
+def install_lib(platform, sof_output_dir, abs_build_dir):
+	global signing_key
+	with os.scandir(str(abs_build_dir)) as iter:
+		for entry in iter:
+			if (not entry.is_dir or
+			    not entry.name.endswith('_llext')):
+				continue
+			uuids = pathlib.Path(entry.path) / 'llext.uuid'
+			if not os.path.exists(uuids):
+				continue
+
+			# replace '_llext' with '.llext', e.g.
+			# eq_iir_llext/eq_iir.llext
+			llext_base = entry.name[:-6]
+			llext_file = llext_base + '.llext'
+
+			sof_lib_dir = pathlib.Path(sof_output_dir / '..' / 'sof-ipc4-lib' / platform)
+			if args.key_type_subdir != "none":
+				sof_lib_dir = sof_lib_dir / args.key_type_subdir
+			sof_lib_dir.mkdir(parents=True, exist_ok=True)
+			dst = sof_lib_dir / llext_file
+			rimage_executable = shutil.which("rimage", path=RIMAGE_BUILD_DIR)
+			rimage_cfg = pathlib.Path(entry.path) / 'rimage_config.toml'
+			llext_input = pathlib.Path(entry.path) / str(llext_base + '.so')
+			llext_output = pathlib.Path(entry.path) / llext_file
+
+			sign_cmd = [str(rimage_executable), "-o", str(llext_output),
+				    "-e", "-c", str(rimage_cfg),
+				    "-k", str(signing_key), "-l", "-r",
+				    str(llext_input)]
```

Comment:
You can't hardcode rimage.path and arguments like that, the user has the ability to override those, see `rimage_west_configuration()`. See top-level comment.

Reply:
Similar to CMake, I still can't see what is temporary versus not here. Please add the relevant TODOs. Even more important considering the high-level signing design sounds like "do whatever you want".

Reply:
@marc-hb everything is temporary... I'm far from being a Python expert, so I expect improvements throughout this code - both during this review and after this is (hopefully soon) merged

Reply:
> everything is temporary...

Yes of course, we all die in the end. Yet there is a huge difference between:

1. This code MAY change some day - or not, who knows?
2. We SHOULD change this ASAP and move it to Zephyr, using new APIx X, Z and Z.

Seeing comment 2. stops people from wasting too much time with the corresponding code.